OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -64,8 +67,16 @@ protected virtual Activity InnerTick(Actor self, AttackBase attack)
 			if (!attack.Info.IgnoresVisibility && type == TargetType.Actor && !Target.Actor.Info.HasTraitInfo<FrozenUnderFogInfo>() && !self.Owner.CanTargetActor(Target.Actor))
 				return NextActivity;
 
+			// Drop the target once none of the weapons are effective against it
+			var armaments = attack.ChooseArmamentsForTarget(Target, forceAttack);
+			if (!armaments.Any())
+				return NextActivity;
+
+			// Update ranges
+			minRange = armaments.MaxBy(a => a.Weapon.MinRange.LengthSquared).Weapon.MinRange;
+			maxRange = armaments.MinBy(a => a.Weapon.Range.LengthSquared).MaxRange();
```

Comment:
You need to order by `a.MaxRange()`, which includes range modifiers, instead of `a.Weapon.Range`.